pkg/drivers/hyperv/hyperv.go

```diff
@@ -87,12 +87,12 @@ func (d *Driver) DriverName() string {
 func (d *Driver) GetState() (state.State, error) {
 	stdout, err := cmdOut("(", "Hyper-V\\Get-VM", d.MachineName, ").state")
 	if err != nil {
-		return state.None, fmt.Errorf("Failed to find the VM status")
+		return state.Error, fmt.Errorf("Failed to find the VM status")
 	}
 
 	resp := parseLines(stdout)
 	if len(resp) < 1 {
-		return state.None, nil
+		return state.Error, fmt.Errorf("unexpected Hyper-V state %s", resp)
```

Comment:
Fwiw, the libvirt machine driver also uses `state.None` in similar situations, but it returns errors. The returned state should be assumed to be invalid anyway if the returned error is not `nil.

Reply:
yes for libvirt I have this one: https://github.com/code-ready/machine-driver-libvirt/pull/88